src/main/scala/dpla/batch_process_dpla_index/processes/ThumbnailPurge.scala

```diff
@@ -0,0 +1,38 @@
+package dpla.batch_process_dpla_index.processes
+
+import com.databricks.spark.avro._
+import dpla.batch_process_dpla_index.helpers.S3FileHelper
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{Column, SparkSession}
+
+object ThumbnailPurge extends S3FileHelper {
+
+  def execute(spark: SparkSession, input: String): String = {
+
+    val df = spark.read.avro(input)
+
+    val thumbnailBucket = "dpla-thumbnails"
+    df.createOrReplaceTempView("df")
+
+    val thumbnailKeys = spark.sql("SELECT df.dplaUri from df")
+      .withColumn("dplaId", regexp_replace(col("dplaUri"), "http://dp.la/api/items/", ""))
+      .select("dplaId")
+      .collect()
```

Comment:
do you need to collect()? wouldn't this go faster if it was on the worker?

Reply:
I could move the `collect` after map but I don't think the actual delete would benefit from running on the workers..delete needs an array, and works in batches  val groupedKeys = keys.grouped(1000)

and IMO this isn't an entry that would get run on a cluster in most cases unlike the other entries in this project.

Reply:
Gotcha!